### PR TITLE
Update pattern docs

### PR DESCRIPTION
**Pull request by Licenser**
Add tuple pattern explanation and a user intention focused description to
each of the patterns: array, tuple, and record

### CODE REVIEW

File: docs/tremor-script/index.md [hunk 1]
```diff
@@ -476,8 +476,44 @@ match sneaky_json of
 end;
 ```
 
+#### Matching tuple patterns
+
+!!! tip
+    A *tuple pattern* matches a *target* value if the *target* is an array and **each** test matches the positionally correspondent value in the *target*. The *target* needs to be **at least as long** as the *pattern* but **can be longer** if the *pattern* ends with `...`.
```
Comment (mfelsche) [inline docs/tremor-script/index.md:482]:
```suggestion
    A *tuple pattern* matches a *target* value if the *target* is an array and **each** test matches the positionally correspondent value in the *target*. The *target* needs to be **at least as long** as the *pattern* but **can be longer** if the *pattern* ends with `..`.
```

File: docs/tremor-script/index.md [hunk 2]
```diff
@@ -488,7 +524,7 @@ Array Pattern filter grammar:
 
 > ![array filter grammar](grammar/diagram/ArrayPatternFilter.png)
 
-In addition to literal array matching, where the case expression array literal must exactly match the target of the match expression, array patterns enable testing for matching elements within an array and filtering on the basis of matched elements.
+In addition to a subset match, where the elements if the pattern must be included in the match the target of the match expression, array patterns enable testing for matching elements within an array and filtering on the basis of matched elements.
```
Comment (mfelsche) [inline docs/tremor-script/index.md:527]:
```suggestion
In addition to a subset match, where the elements of the pattern must be included in the target of the match expression, array patterns enable testing for matching elements within an array and filtering on the basis of matched elements.
```

**Review by mfelsche — APPROVED** [review]
some small typos.

File: docs/tremor-script/index.md (hunk 1 above)
Comment (Licenser) [reply docs/tremor-script/index.md:482]: it's 3 dots :D

**Review by mfelsche — APPROVED** [review]

**Pull request merged**